integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/config/IntegrasjonspunktBeans.java

```diff
@@ -1,134 +1,190 @@
+        StatusStrategyFactory statusStrategyFactory = new StatusStrategyFactory();
+        statusStrategies.forEach(statusStrategyFactory::registerStrategy);
+        return statusStrategyFactory;
     }
 
-    @ConditionalOnProperty(name="difi.move.feature.enableDPF", havingValue = "true")
     @Bean
-    public FiksStatusStrategy fiksConversationStrategy(SvarUtService svarUtService, ConversationService conversationService) {
-        return new FiksStatusStrategy(svarUtService, conversationService);
+    public DpiReceiptService dpiReceiptService(IntegrasjonspunktProperties integrasjonspunktProperties, MeldingsformidlerClient meldingsformidlerClient) {
+        return new DpiReceiptService(integrasjonspunktProperties, meldingsformidlerClient);
+    }
+
+    @Bean(name = "fiksMailClient")
+    public NoarkClient fiksMailClient(IntegrasjonspunktProperties properties) {
+        return new MailClient(properties, properties.getFiks().getInn().getMailSubject());
     }
 
     @Bean
-    public StatusStrategyFactory statusStrategyFactory(List<StatusStrategy> statusStrategies) {
-        StatusStrategyFactory statusStrategyFactory = new StatusStrategyFactory();
-        statusStrategies.forEach(statusStrategyFactory::registerStrategy);
-        return statusStrategyFactory;
+    public JWTDecoder jwtDecoder() throws CertificateException {
+        return new JWTDecoder();
     }
 
     @Bean
-    public StrategyFactory messageStrategyFactory(MessageSender messageSender,
-                                                  ServiceRegistryLookup serviceRegistryLookup,
-                                                  KeystoreProvider meldingsformidlerKeystoreProvider,
-                                                  @Lazy InternalQueue internalQueue,
-                                                  @Qualifier("localNoark") ObjectProvider<NoarkClient> localNoark,
-                                                  @SuppressWarnings("SpringJavaAutowiringInspection") ObjectProvider<List<MessageStrategyFactory>> messageStrategyFactory) {
-        final StrategyFactory strategyFactory = new StrategyFactory(messageSender, serviceRegistryLookup, meldingsformidlerKeystoreProvider, properties, localNoark.getIfAvailable(), internalQueue);
-        if(messageStrategyFactory.getIfAvailable() != null) {
-            messageStrategyFactory.getIfAvailable().forEach(strategyFactory::registerMessageStrategyFactory);
-        }
-        return strategyFactory;
+    @Scope(value = ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+    public CmsUtil cmsUtil() {
+        return new CmsUtil();
     }
 
     @Bean
-    public DpiReceiptService dpiReceiptService(IntegrasjonspunktProperties integrasjonspunktProperties, KeystoreProvider keystoreProvider) {
-        return new DpiReceiptService(integrasjonspunktProperties, keystoreProvider);
+    public PublicKey publicKey(IntegrasjonspunktProperties props) throws CertificateException, IOException {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this method to throw at most one checked exception instead of: java.security.cert.CertificateException, java.io.IOException [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1160)